Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+# DocFX Document Schema Design Specification
```

Comment:
This can be not limited to API

---
In reply to: [122867233](https://github.com/dotnet/docfx/pull/1771#discussion_r122867233) [](ancestors = 122867233)